GameServerLib/Logic/Content/NavGrid.cs

```diff
@@ -30,7 +32,52 @@ public class NavGrid : INavGrid
         public float MapHeight { get; set; }
         public Vector2 MiddleOfMap { get; set; }
         public const float SCALE = 2f;
+        private Grid grid;
 
+        public void InitializePathfinding()
+        {
+            grid = new Grid((int)XCellCount, (int)YCellCount);
+            foreach (var cell in Cells)
+            {
+                if (cell.HasFlag(this, NavigationGridCellFlags.NOT_PASSABLE))
+                {
+                    grid.BlockCell(new Position(cell.X, cell.Y));
+                }
+            }
+        }
+
+        public List<Vector2> GetPath(Vector2 from, Vector2 to)
+        {
+            List<Vector2> returnList = new List<Vector2>() { from };
+            try
+            {
+                var vectorFrom = TranslateToNavGrid(new Vector<float> { X = from.X, Y = from.Y });
+                var cellFrom = GetCell((short)vectorFrom.X, (short)vectorFrom.Y);
+
+                var vectorTo = TranslateToNavGrid(new Vector<float> { X = to.X, Y = to.Y });
+                var cellTo = GetCell((short)vectorTo.X, (short)vectorTo.Y);
+
+                if(cellFrom != null && cellTo != null)
+                {
+                    Position[] path = grid.GetPath(new Position(cellFrom.X, cellFrom.Y), new Position(cellTo.X, cellTo.Y));
+                    if (path != null)
+                    {
+                        foreach (var position in path)
+                        {
+                            var navGridCell = GetCell(position.X, position.Y);
+                            var cellPosition = TranslateFromNavGrid(new Vector<float>() { X = navGridCell.X, Y = navGridCell.Y });
+                            returnList.Add(new Vector2(cellPosition.X, cellPosition.Y));
+                        }
+                    }
+                }                
+            }
+            catch (Exception ex)
+            {
+
```

Comment:
Is there a specific exception we should catch?
If this does break, shouldn't we be logging that a problem occurred?
If this is to just catch bad cell positions, we should fix this with checks rather than using a try catch.